tools/google-cloud-support-slackbot/google_cloud_support_slackbot.py

```diff
@@ -0,0 +1,811 @@
+    comments from the previous pull. If any change is detected between the two versions 
+    of the case, the change is posted to any channel that is tracking it
+    """
+    
+    query_string = 'organization="organizations/{}" AND state=OPEN'.format(ORG_ID)
+
+    while True:
+        loop_skip = False
+        sleep_timer = 15
+        cases = {}
+        try:
+            if os.path.exists(cases_file):
+                with open(cases_file) as f:
+                    cases = json.load(f)
+        except Exception as e:
+            logging.error(e, ' : {}'.format(datetime.now()))
+            pass
+
+        req = support_service.cases().search(query=query_string)
+        try:
+            resp = req.execute().get('cases', [])
+        except Exception as e:
+            logging.error(e, ' : {}'.format(datetime.now()))
+            time.sleep(5)
+            continue
+
+        temp_cases = {}
+        
+        for case in resp:
+            try:
+                temp_case = Support_Case(case)
+                temp_cases[temp_case.case_number] = temp_case
+            except Exception as e:
+                logging.error(e, ' : {}'.format(datetime.now()))
+                loop_skip = True
+                break
+
+        if loop_skip == True:
+            time.sleep(5)
+            continue
+
+        # Remove the cases from our dictionary that are no longer active
+        for key in cases:
+            delete_entry = True
+            for temp_key in temp_cases:
+                if key == temp_key:
+                    delete_entry = False
+                    break
+            if delete_entry == True:
+                notify_slack(key, 'closed', '')
+                del cases[key]
+        
+        # Check for existing cases that have a new update time. Post their relevant update
+        # to the channels that are tracking those cases.
+        for temp_key in temp_cases:
+            if temp_key in cases:
+                if cases[temp_key]['comment_list'] != temp_cases[temp_key].comment_list:
+                    if 'googleSupport' in temp_cases[temp_key].comment_list[0]['creator']:
+                        notify_slack(temp_key, 'comment', temp_cases[temp_key].comment_list[0]['body'])
+                if cases[temp_key]['priority'] != temp_cases[temp_key].priority:
+                    notify_slack(temp_key, 'priority', temp_cases[temp_key].priority)
+                if cases[temp_key]['escalated'] != temp_cases[temp_key].escalated:
+                    if temp_cases[temp_key].escalated == True:
+                        notify_slack(temp_key, 'escalated', temp_cases[temp_key].escalated)
+                    else:
+                        notify_slack(temp_key, 'de-escalated', temp_cases[temp_key].escalated)
+        
+        # Replace the stored case list with our latest pull and update the file
+        try:
+            file_overwrite(cases_file, temp_cases)        
+        except Exception as e:
```

Comment:
What specific exception are you capturing here? Please narrow it down to the specific situations that you are trying to handle, this is too broad.